ddtrace/tracer/span_event_test.go

```diff
@@ -163,3 +168,152 @@ func Test_spanAddEvent(t *testing.T) {
 		assert.Nil(t, evt.RawAttributes)
 	})
 }
+
+type CustomError struct {
+	Code    int
+	Message string
+}
+
+func (e CustomError) Error() string {
+	return fmt.Sprintf("custom error %d: %s", e.Code, e.Message)
+}
+
+func Test_recordException(t *testing.T) {
+	t.Run("RecordException multiple times", func(t *testing.T) {
+		s := newBasicSpan("test")
+		s.supportsEvents = true
+
+		// Record multiple exceptions
+		err1 := errors.New("first error")
+		err2 := errors.New("second error")
+
+		s.RecordException(err1)
+		s.RecordException(err2)
+
+		s.Finish()
+
+		require.Len(t, s.spanEvents, 2)
+
+		// Check first exception
+		evt1 := s.spanEvents[0]
+		assert.Equal(t, "exception", evt1.Name)
+		assert.Equal(t, 3, len(evt1.Attributes))
+		assert.Equal(t, "*errors.errorString", evt1.Attributes["exception.type"].StringValue)
+		assert.Equal(t, "first error", evt1.Attributes["exception.message"].StringValue)
+		assert.NotEmpty(t, evt1.Attributes["exception.stacktrace"].StringValue)
+
+		// Check second exception
+		evt2 := s.spanEvents[1]
+		assert.Equal(t, "exception", evt2.Name)
+		assert.Equal(t, 3, len(evt2.Attributes))
+		assert.Equal(t, "*errors.errorString", evt2.Attributes["exception.type"].StringValue)
+		assert.Equal(t, "second error", evt2.Attributes["exception.message"].StringValue)
+		assert.NotEmpty(t, evt2.Attributes["exception.stacktrace"].StringValue)
+	})
+
+	t.Run("RecordException with custom attributes", func(t *testing.T) {
```

Comment:
Would be nice to see an example of a custom attribute overwriting a default attribute here as well!